web/packages/teleterm/src/ui/Vnet/DiagnosticsAlert.tsx

```diff
@@ -0,0 +1,181 @@
+    ? {
+        disabled: true,
+        title: 'Log in to a cluster to see the full report',
+      }
+    : {};
+  const openReport = () => {
+    // TODO(ravicious): Open report doc.
+  };
+
+  if (
+    report.checkResults.length &&
+    report.checkResults.every(
+      checkResult =>
+        checkResult.status === CheckStatus.OK &&
+        checkResult.checkReport.status === CheckReportStatus.OK
+    )
+  ) {
+    // TODO(ravicious): Once we start automatically running checks, this alert needs to be shown
+    // only if the user manually requested diagnostics to be run. Alternatively, we can replace it
+    // with some kind of a smaller "Everything's okay" indicator, but the user needs to be able to
+    // open the report anyway.
+    return (
+      <SliderStepAlert
+        kind="success"
+        onDismiss={onDismiss}
+        buttons={
+          <ActionButton
+            fill="border"
+            intent="neutral"
+            inputAlignment
+            action={{ content: 'Open Report', onClick: openReport }}
+            {...disabledOpenReportButtonProps}
+          />
+        }
+      >
+        No issues found.
+      </SliderStepAlert>
+    );
+  }
+
+  // If this default warningText is shown the user, it means we failed to account for a specific
+  // state.
+  let warningText = 'Unknown report status';
+  if (
+    report.checkResults.some(
+      checkResult =>
+        checkResult.status === CheckStatus.OK &&
+        checkResult.checkReport.status === CheckReportStatus.ISSUES_FOUND
+    )
+  ) {
+    warningText = 'Other software on your device might interfere with VNet.';
+  } else if (
+    report.checkResults.some(
+      checkResult => checkResult.status === CheckStatus.ERROR
+    )
+  ) {
+    warningText = 'Some diagnostic checks failed to report results.';
+  }
+
+  return (
+    <SliderStepAlert
+      kind="warning"
+      onDismiss={onDismiss}
+      buttons={
+        <>
+          <ActionButton
+            fill="border"
+            intent="neutral"
+            inputAlignment
+            action={{ content: 'Open Report', onClick: openReport }}
+            {...disabledOpenReportButtonProps}
+          />
+          <ActionButton
+            fill="minimal"
+            intent="neutral"
+            inputAlignment
+            action={{ content: 'Retry', onClick: runDiagnostics }}
+          />
+        </>
+      }
+    >
+      {warningText}
+    </SliderStepAlert>
+  );
+};
+
+const SliderStepAlert = (
+  props: (
+    | { buttons?: ReactNode; details?: never }
+    | { details?: ReactNode; buttons?: never }
+  ) &
+    Required<Pick<AlertProps, 'kind' | 'onDismiss' | 'children'>>
+) => {
+  const { buttons, details, ...alertProps } = props;
+
+  return (
+    <Alert
+      mt={0}
+      mx={textSpacing}
+      mb={textSpacing}
+      dismissible
+      alignItems={buttons ? 'flex-start' : 'center'}
+      details={
+        buttons ? (
+          <Flex mt={2} gap={2}>
+            {buttons}
```

Comment:
I couldn't use `primaryAction` and `secondaryAction` available on `<Alert>`, as buttons for them are placed to the right of alert text. There's no space for that in the VNet panel, so I decided to misuse `details` for that.

If there's more places which could use this approach, we could think of codifying this as an `Alert` prop. But for now I didn't want to refactor `Alert` significantly just to support a single custom use case.